src/Fixer/PhpUnit/PhpUnitDedicateAssertFixer.php

```diff
@@ -625,4 +708,39 @@ private function cloneAndClearTokens(Tokens $tokens, int $start, int $end): arra
 
         return $clone;
     }
+
+    /**
+     * @param array{
+     *     index: int,
+     *     loweredName: string,
+     *     openBraceIndex: int,
+     *     closeBraceIndex: int,
+     * } $assertCall
+     */
+    private function fixAssertNewNames(Tokens $tokens, array $assertCall): void
+    {
+        if (!\in_array($assertCall['loweredName'], $this->functions, true)) {
+            return;
+        }
+
+        $replacement = null;
+        if ('assertnotisreadable' === $assertCall['loweredName']) {
+            $replacement = 'assertIsNotReadable';
+        } elseif ('assertnotiswritable' === $assertCall['loweredName']) {
+            $replacement = 'assertIsNotWritable';
+        } elseif ('assertdirectorynotexists' === $assertCall['loweredName']) {
+            $replacement = 'assertDirectoryDoesNotExist';
+        } elseif ('assertfilenotexists' === $assertCall['loweredName']) {
+            $replacement = 'assertFileDoesNotExist';
+        }
```

Comment:
Strange that mess-detector prefers this over switch statement, not everything can be refactored to polymorphism, especially string comparison.

Reply:
it complains because switch-case uses `==` and not `===`. It is a good it's complaining.

Reply:
```php
        $replacements = [
            'assertnotisreadable' => 'assertIsNotReadable',
            'assertnotiswritable' => 'assertIsNotWritable',
            'assertdirectorynotexists' => 'assertDirectoryDoesNotExist',
            'assertfilenotexists' => 'assertFileDoesNotExist',
        ];
        $replacement = $replacements[$assertCall['loweredName']] ?? null;
```

(we can't use `match` yet)